src/Network/NWEthernetChannel.cs

```diff
@@ -0,0 +1,208 @@
+				del (err);
+				err?.Dispose ();
+			}
+		}
+
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public void Send (ReadOnlySpan<byte> content, ushort vlanTag, string remoteAddress, Action<NWError> callback)
+		{
+			if (callback == null)
+				throw new ArgumentNullException (nameof (callback));
+
+			using (var dispatchData = DispatchData.FromByteBuffer (content.ToArray (), 0, content.Length)) {
+				BlockLiteral block_handler = new BlockLiteral ();
+				block_handler.SetupBlockUnsafe (static_SendCompletion, callback);
+
+				try {
+					nw_ethernet_channel_send (GetCheckedHandle (), dispatchData.GetHandle (), vlanTag, remoteAddress, ref block_handler);
+				} finally {
+					block_handler.CleanupBlock ();
+				}
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		unsafe static extern void nw_ethernet_channel_set_receive_handler (OS_nw_ethernet_channel ethernet_channel, /* [NullAllowed] */ void *handler);
+
+		delegate void nw_browser_set_state_changed_handler_t (IntPtr block, OS_dispatch_data content, ushort vlan_tag, byte[] local_address, byte[] remote_address);
+		static nw_browser_set_state_changed_handler_t static_ReceiveHandler = TrampolineReceiveHandler;
+
+		[MonoPInvokeCallback (typeof (nw_browser_set_state_changed_handler_t))]
+		static void TrampolineReceiveHandler (IntPtr block, OS_dispatch_data content, ushort vlanTag, byte[] localAddress, byte[] remoteAddress)
+		{
+			var del = BlockLiteral.GetTarget<NWEthernetChannelReceiveDelegate> (block);
+			if (del != null) {
+				DispatchData dispatchData = null, dataCopy = null;
+				IntPtr bufferAddress = IntPtr.Zero;
+				nuint bufferSize = 0;
+
+				if (content != IntPtr.Zero) {
+					dispatchData = new DispatchData (content, owns: false);
+					dataCopy = dispatchData.CreateMap (out bufferAddress, out bufferSize);
+				}
+
+				unsafe {
+					var spanData = new ReadOnlySpan<byte> ((void*)bufferAddress, (int)bufferSize);
+					del (spanData, vlanTag, Encoding.UTF8.GetString (localAddress), Encoding.UTF8.GetString (remoteAddress));
+				}
+
+				if (dispatchData != null) {
+					dataCopy.Dispose ();
+					dispatchData.Dispose ();
+				}
+			}
+		}
+
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public void SetReceiveHandler (NWEthernetChannelReceiveDelegate handler)
+		{
+			unsafe
+			{
+				if (handler == null) {
+					nw_ethernet_channel_set_receive_handler (GetCheckedHandle (), null);
+					return;
+				}
+
+				BlockLiteral block_handler = new BlockLiteral ();
+				BlockLiteral *block_ptr_handler = &block_handler;
+				block_handler.SetupBlockUnsafe (static_ReceiveHandler, handler);
+				try {
+					nw_ethernet_channel_set_receive_handler (GetCheckedHandle (), (void*) block_ptr_handler);
+				} finally {
+					block_handler.CleanupBlock ();
+				}
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		unsafe static extern void nw_ethernet_channel_set_state_changed_handler (OS_nw_ethernet_channel ethernet_channel, /* [NullAllowed] */ void *handler);
```

Comment:
More exact type is always welcome:

```suggestion
		unsafe static extern void nw_ethernet_channel_set_state_changed_handler (OS_nw_ethernet_channel ethernet_channel, /* [NullAllowed] */ BlockLiteral *handler);
```